app/src/main/java/com/x8bit/bitwarden/data/autofill/parser/AutofillParserImpl.kt

```diff
@@ -64,20 +64,34 @@ class AutofillParserImpl(
     /**
      * Parse the [AssistStructure] into an [AutofillRequest].
      */
+    @Suppress("LongMethod")
     private fun parseInternal(
         assistStructure: AssistStructure,
         autofillAppInfo: AutofillAppInfo,
         fillRequest: FillRequest?,
     ): AutofillRequest {
         // Parse the `assistStructure` into internal models.
         val traversalDataList = assistStructure.traverse()
-        // Flatten the autofill views for processing.
+        // Take only the autofill views from the node that currently has focus.
+        // Then remove all the fields that cannot be filled with data.
+        // We fallback to taking all the views if nothing has focus.
         val autofillViews = traversalDataList
-            .map { it.autofillViews }
+            .mapNotNull { traversalNode ->
+                traversalNode.autofillViews.takeIf { views -> views.any { it.data.isFocused } }
+            }
```

Comment:
This works but I wonder if this is slightly easier to follow if we split apart the mapping from the filtering:

```kotlin
        val autofillViews = traversalDataList
            .map { it.autofillViews }
            .filter { views -> views.any { it.data.isFocused } }
            .flatten()
            .filter { it !is AutofillView.Unused }
            ?: traversalDataList
                .map { it.autofillViews }
                .flatten()
                .filter { it !is AutofillView.Unused }
```

Reply:
I  elected not to do that just to avoid another loop through the list, I was valuing performance over readability here. What do you think about that?